### PR TITLE
feat: 🍰 My Groups Page

### LINKED ISSUE #5140: 🚀 [Feature] My Groups Page
## :rocket: Feature Request

My Groups Page

My Groups Page [#](https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/5059#user-content-my-groups-page)
Reach the my groups page under the path domain.social/mygroups.

EPIC #5059 

### My Groups Page <a href="#user-content-my-groups-page" id="my-groups-page">#</a>

Reach the `my groups` page under the path `domain.social/mygroups`.

On this page you find:

- One list of the groups you own (maybe administer) and where you are a member.
  - Have visual sign for your owned groups.
- An add button to found a new group.

List view:

- It's a list of group cards. (like our post cards)
- A group card shows a short form of the group informations.

Functionallities of a group card: (list item)

- Click on group card links to the [group profile](#user-content-group-profile-page).
  - On the group profile you can reach the [group settings](#user-content-group-settings) via the contect menu.
- (later?) Context menu directly on the post card
  - As it is on the [group profile](#user-content-group-profile-page).

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest

Preparation for 
- Settings , My groups
- Manage groups
- Create new groups

### Issues
 - fixes #5140 
 
### Todo

- [x] shall path be `group/mygroups` or just `mygroups`?
- [x] deconstruction of mutation `variables` in `create.vue`

### CODE REVIEW

File: webapp/components/Group/GroupCard.vue [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<template>
+  <div>
+    <ds-container class="group-card">
+      <ds-page-title heading="Groups"></ds-page-title>
+      <ds-card v-for="item in items" :key="item.id" space="xx-small">
+        <nuxt-link to="/group/g1/testgruppe">{{ item.name }}</nuxt-link>
```
Comment (Tirokk) [inline webapp/components/Group/GroupCard.vue:6]: What is this?

File: webapp/components/Group/GroupForm.vue [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+      formData: {
+        name: name || '',
+        groupType: groupType || '',
+        about: about || '',
+        description: description || '',
+        actionRadius: actionRadius || '',
+        categoryIds: categories ? categories.map((category) => category.id) : [],
+      },
+      formSchema: {
+        name: { required: true, min: 3, max: 100 },
```
Comment (Tirokk) [inline webapp/components/Group/GroupForm.vue:104]: Please use constants from a constants file for `min` and `max`.

File: webapp/components/Group/GroupForm.vue [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        groupType: { required: true },
+        about: { required: true },
+        description: { required: true },
+        actionRadius: { required: true },
+        categoryIds: {
+          type: 'array',
+          required: this.categoriesActive,
+          validator: (_, value = []) => {
+            if (this.categoriesActive && (value.length === 0 || value.length > 3)) {
```
Comment (Tirokk) [inline webapp/components/Group/GroupForm.vue:113]: We already have constants in file `webapp/constants/categories.js` for min and max.

File: webapp/components/Group/GroupForm.vue [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+          })
+        : this.$emit('createGroup', {
+            name,
+            about,
+            description,
+            groupType,
+            actionRadius,
+            categoryIds,
+          })
+    },
```
Comment (Tirokk) [inline webapp/components/Group/GroupForm.vue:143]:
```suggestion
      const variables = {
        name,
        about,
        description,
        groupType,
        actionRadius,
        categoryIds,
      }
      this.update
        ? this.$emit('updateGroup', {
            ...variables,
            id: this.group.id,
          })
        : this.$emit('createGroup', variables)
    },
```

File: webapp/pages/group/edit/_id.vue [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            // const { sendNotificationEmails } = createGroup
+            // this.setCreateGroup({
+            //   ...this.createGroup,
+            //   sendNotificationEmails,
+            // })
+            this.$toast.success(this.$t('group.group-updated'))
+          },
+        })
+      } catch (error) {
+        // this.notifyByEmail = !this.notifyByEmail
```
Comment (Tirokk) [inline webapp/pages/group/edit/_id.vue:75]: Where this is comming from?

File: webapp/pages/group/edit/_id/index.vue [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+            if (this.categoriesActive && (value.length === 0 || value.length > 3)) {
+              return [new Error(this.$t('common.validations.categories'))]
+            }
+            return []
+          },
+        },
+      },
+    }
+  },
+
```
Comment (Tirokk) [inline webapp/pages/group/edit/_id/index.vue:122]:
```suggestion
```

File: webapp/pages/settings.vue [hunk 7]
```diff
@@ -39,6 +39,10 @@ export default {
           name: this.$t('settings.social-media.name'),
           path: `/settings/my-social-media`,
         },
+        {
+          name: this.$t('settings.myGroups'),
+          path: `/group/my-groups`,
+        },
```
Comment (Tirokk) [inline webapp/pages/settings.vue:45]:
```suggestion
        {
          name: this.$t('settings.myGroups'),
          path: `/my-groups`,
        },
```

I remember that we liked to use this suggested route for the list of my groups.

**Review by Tirokk — CHANGES_REQUESTED** [review]
Cool !!! 🤗🍀

I have some suggestions …

File: webapp/pages/group/create.vue [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    return {
+      createGroupData: {},
+    }
+  },
+  methods: {
+    async createGroup(value) {
+      try {
+        await this.$apollo.mutate({
+          mutation: createGroupMutation,
+          variables: value,
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/pages/group/create.vue:35]: done

File: webapp/pages/group/create.vue [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+            // const { sendNotificationEmails } = createGroup
+            // this.setCreateGroup({
+            //   ...this.createGroup,
+            //   sendNotificationEmails,
+            // })
+            this.$toast.success(this.$t('group.group-created'))
+          },
+        })
+      } catch (error) {
+        // this.notifyByEmail = !this.notifyByEmail
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/pages/group/create.vue:46]: deleted

File: webapp/pages/group/edit/_id.vue (hunk 5 above)
Comment (ogerly) [reply webapp/pages/group/edit/_id.vue:75]: deleted

File: webapp/pages/group/edit/_id/members.vue [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+          name: 'Twitch',
+          type: 'ADC',
+          loves: 'Spray and pray',
+        },
+        {
+          name: 'Blitz',
+          type: 'Support',
+          loves: 'Hook you up',
+        },
+      ],
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/pages/group/edit/_id/members.vue:44]: done

File: webapp/pages/settings.vue (hunk 7 above)
Comment (ogerly) [reply webapp/pages/settings.vue:45]: done

File: webapp/pages/group/edit/_id/index.vue [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+            id: this.group.id,
+          })
+        : this.$emit('createGroup', {
+            name,
+            about,
+            description,
+            groupType,
+            actionRadius,
+            categoryIds,
+          })
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/pages/group/edit/_id/index.vue:141]: done

File: webapp/components/Group/GroupCard.vue (hunk 1 above)
Comment (ogerly) [reply webapp/components/Group/GroupCard.vue:6]: done

File: webapp/components/Group/GroupForm.vue (hunk 2 above)
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:104]: done

File: webapp/components/Group/GroupForm.vue (hunk 3 above)
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:113]: done

File: webapp/pages/group/edit/_id.vue [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+      try {
+        await this.$apollo.mutate({
+          mutation: updateGroupMutation,
+          variables: value,
+          update: (_, { data: { updateGroupData } }) => {
+            // const { sendNotificationEmails } = createGroup
+            // this.setCreateGroup({
+            //   ...this.createGroup,
+            //   sendNotificationEmails,
+            // })
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/pages/group/edit/_id.vue:70]: done

File: webapp/components/Group/GroupForm.vue [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+          v-model="formData.name"
+          label="Gruppenname"
+          placeholder="Your name ..."
+        ></ds-input>
+
+        <ds-select
+          icon="user"
+          v-model="formData.groupType"
+          label="Sichtbarkeit"
+          :options="['public', 'close', 'hidden']"
```
Comment (Tirokk) [inline webapp/components/Group/GroupForm.vue:21]:
```suggestion
          :options="['public', 'closed', 'hidden']"
```

File: webapp/pages/my-groups.vue [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+    <div>my groups</div>
+    <group-teaser />
+    <br />
+    <br />
+    <group-card :items="responseGroupListQuery" />
+  </div>
+</template>
+<script>
+import GroupTeaser from '~/components/Group/GroupTeaser.vue'
+import GroupCard from '~/components/Group/GroupCard.vue'
```
Comment (Tirokk) [inline webapp/pages/my-groups.vue:12]: `GroupTeaser` and `GroupCard` is not named correctly here, but we can care for that later on …

**Review by Tirokk — CHANGES_REQUESTED** [review]
I found one little mistake @ogerly .
And the tests found locales to be fixed …

I fixed both!
Hope that's okay?

And just a comment for the future.

**Review by Tirokk — APPROVED** [review]
A very cool beginning @ogerly . 😍

This are really great features we started with. 🚀🚀💫💫

**Pull request merged**